redfish-core/lib/log_services.hpp

```diff
@@ -3411,6 +3436,610 @@ inline void getSystemHardwareIsolationLogService(
+{
+    std::string isolatedHwItemId;
+
+    if ((dbusObjPath.filename().find("cpu") != std::string::npos) &&
+        (dbusObjPath.parent_path().filename().find("dcm") != std::string::npos))
+    {
+        isolatedHwItemId = std::string(dbusObjPath.parent_path().filename() +
+                                       "-" + dbusObjPath.filename());
+    }
+    else
+    {
+        isolatedHwItemId = dbusObjPath.filename();
+    }
+    return isolatedHwItemId;
+}
+
+/**
+ * @brief API used to fill the respective redfish resource "Id" in the given
+ *        redfishUri by using the given dbus object path which represents the
+ *        given redfishUri and assign the filled redfish uri into
+ *        "OriginOfCondition" property of LogEntry schema.
+ *
+ * @param[in] asyncResp - The redfish response to return.
+ * @param[in] entryJsonIdx - The json entry index to add isolated hardware
+ *                            details in the appropriate entry json object.
+ * @param[in] dbusObjPath - The DBus object path which represents redfishUri.
+ * @param[in] redfishUri - The redfish uri to fill with its "Id".
+ *
+ * @return The redfish response with "OriginOfCondition" property of
+ *         LogEntry schema if success else nothing in redfish response.
+ *
+ * @note - This function will get the associated redfish uri prefix of "Id"
+ *         ("<str>") and look at the predefined redfish list (which will
+ *         contains the pair <DBus Interfaces (which represents redfish
+ *         resource), RedfishUri> of elements) to get the dbus interfaces and
+ *         that is used to get the dbus objects which will compare with the
+ *         given dbus object path to fill the respective uri Id in the
+ *         given redfishUri.
+ *       - This function will run recursively until filling all Ids in given
+ *         given redfishUri.
+ */
+inline void
+    fillRedfishUriWithIds(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                          const size_t entryJsonIdx,
+                          const sdbusplus::message::object_path& dbusObjPath,
+                          std::string& redfishUri)
+{
+
+    std::string uriIdPattern{"<str>"};
+    if (auto uriIdPos = redfishUri.rfind(uriIdPattern);
+        uriIdPos != std::string::npos)
+    {
+        // Reduce by "1" from uriIdPos to get the redfish uri of uriIdPattern
+        // For example,
+        // "/redfish/v1/Systems/system/Processors/<str>/SubProcessors/core0"
+        //                                       ^
+        auto parentRedfishUri = redfishUri.substr(0, uriIdPos - 1);
+        auto parentRedfishUriIt =
+            std::find_if(redfishUriList.begin(), redfishUriList.end(),
+                         [parentRedfishUri](const auto& ele) {
+                             return parentRedfishUri == std::get<1>(ele);
+                         });
+
+        if (parentRedfishUriIt == redfishUriList.end())
+        {
+            BMCWEB_LOG_ERROR
```

Comment:
This is not actually a failure right ? When the url list is over, this is used to end the recursive calls